lib/EditableList/EditableListForm.js

```diff
@@ -347,6 +361,48 @@ class EditableListForm extends React.Component {
     this.toggleEdit(index);
   }
 
+  handleDeleteClick = (fields, index) => {
+    const {
+      withDeleteConfirmation,
+      uniqueField,
+    } = this.props;
+
+    if (!withDeleteConfirmation) {
+      this.onDelete(fields, index);
+    }
```

Comment:
It seems that `return` should be added after `onDelete` otherwise the confirmation modal will be shown for both options.